src/fsharp/MethodCalls.fs

```diff
@@ -605,7 +635,10 @@ let CoerceFromFSharpFuncToDelegate g amap infoReader ad callerArgTy m callerArgE
 
 // Handle adhoc argument conversions
 let AdjustCallerArgExprForCoercions (g: TcGlobals) amap infoReader ad isOutArg calledArgTy (reflArgInfo: ReflectedArgInfo) callerArgTy m callerArgExpr = 
-
+  // todo after rebase:
```

Comment:
Recollecting the full context here: all places calling `AdjustCallerArgExprForCoercions` happen to be just deconstructing `CallerArgs` instances into `isOutArg`, `callerArgTy`, `m` and `callerArgExpr` arguments here.

It felt to me like a potential for refactor (remove the individual parameters and pass the `CallerArgs` record instead) but I wasn't 100% clear about it because some call hardcode `isOutArg` to false.

I had the chance to do a review with @TIHan about this idea, and he pointed that `isOutArg` is guaranted to match at those call sites (only happening in `TcSetterArgExpr`) because this is guarded by `error` if the value is `true`.

I think it is better to do that in a separate PR as it is a pure refactor not related to the error message.